### PR TITLE
Add support for evolution-data-server 3.45+ / libsoup-3

### LINKED ISSUE #243: support for evolution-data-server 3.45+ / libsoup-3
Starting with GNOME 43, some core components of the desktop have been ported to libsoup-3, which apparently brings a lot of new features, including HTTP/2 support. However, applications and libraries **cannot** load both libsoup-2.4 and libsoup-3 in the same process, as this is unsupported and will crash the application immediately.

This change indirectly affects switchboard-onlineaccounts, because evolution-data-server switched to libsoup-3 as of version 3.45.1, and that includes a few API changes. Most notably, there's an intentional change in libedataserver-1.2 / `E.SourceWebdav` that breaks builds of this project. I asked on the [Fedora development mailing list](https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/FITD7MFN4V5LXXUFXFXOJTXDTGX7LAQV/) and got this response from the maintainer:

> the SoupURI structure is gone in libsoup3, there is used GUri
> instead. For that reason the ESourceWebdav has replaced the property with the
> "uri" property, using the GUri structure.

I'm not sure how the switch from libsoup-2.4 to libsoup-3 will affect other switchboard plus yet. But I think it will crash switchboard if it tries to load plugins that are linked against different libsoup, so the plugs might need to all add support for and be switched over to using libsoup-3 at the same time.

### PR DESCRIPTION
**Pull request by meisenzahl**
Closes #243

### CODE REVIEW

File: src/Dialogs/CaldavDialog.vala [hunk 1]
```diff
@@ -473,7 +477,11 @@ public class OnlineAccounts.CaldavDialog : Hdy.Window {
                 string? webdav_host = null;
                 if (source.has_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND)) {
                     unowned var webdav_extension = (E.SourceWebdav) source.get_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND);
-                    webdav_host = webdav_extension.soup_uri.host;
+                    #if HAS_EDS_3_45
+                    webdav_host = webdav_extension.uri.get_host ();
+                    #else
+                        webdav_host = webdav_extension.soup_uri.host;
```
Comment (meisenzahl) [inline src/Dialogs/CaldavDialog.vala:483]:
```suggestion
                    webdav_host = webdav_extension.soup_uri.host;
```

File: src/Dialogs/CaldavDialog.vala [hunk 2]
```diff
@@ -397,7 +397,11 @@ public class OnlineAccounts.CaldavDialog : Hdy.Window {
             col.backend_name = "caldav";
 
             unowned var webdav = (E.SourceWebdav)source.get_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND);
+            #if HAS_EDS_3_45
```
Comment (tintou) [inline src/Dialogs/CaldavDialog.vala:400]: Do not indent the macro for better readability and consistency with the rest of our apps :slightly_smiling_face:

File: src/Dialogs/CaldavDialog.vala [hunk 3]
```diff
@@ -550,11 +562,19 @@ public class OnlineAccounts.CaldavDialog : Hdy.Window {
 
         if (collection_source.has_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND)) {
             unowned var webdav_extension = (E.SourceWebdav) collection_source.get_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND);
+            #if HAS_EDS_3_45
+            url_entry.text = webdav_extension.uri.to_string ();
+
+            if (webdav_extension.uri.get_user () != null && webdav_extension.uri.get_user () != "") {
+                url_entry.text = url_entry.text.replace (webdav_extension.uri.get_user () + "@", "");
+            }
```
Comment (tintou) [inline src/Dialogs/CaldavDialog.vala:570]:
```suggestion
            unowned var uri_user = webdav_extension.uri.get_user ();
            if (uri_user != null && uri_user != "") {
                url_entry.text = url_entry.text.replace (uri_user + "@", "");
            }
```

**Comment by meisenzahl** [comment]
@tintou thank you for your feedback!

**Review by tintou — APPROVED** [review]
Looks great to me!

**Pull request merged**